From 4166816bf4c637a075b5b4a5078df415ac5201da Mon Sep 17 00:00:00 2001
From: Ioana Radulescu <ruxandra.radulescu@nxp.com>
Date: Thu, 11 Apr 2019 12:38:28 +0300
Subject: [PATCH 229/235] dpaa2-eth: Fix incorrect version check

The code checking whether the new autoneg API is supported
had a bug, fix it.

Signed-off-by: Catalin Neacsu <valentin-catalin.neacsu@nxp.com>
[Xulin: Original patch taken from NXP LSDK-19.06.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/staging/fsl-dpaa2/ethernet/dpaa2-ethtool.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/staging/fsl-dpaa2/ethernet/dpaa2-ethtool.c b/drivers/staging/fsl-dpaa2/ethernet/dpaa2-ethtool.c
index 330dab6977bd..70209a9d8add 100644
--- a/drivers/staging/fsl-dpaa2/ethernet/dpaa2-ethtool.c
+++ b/drivers/staging/fsl-dpaa2/ethernet/dpaa2-ethtool.c
@@ -180,7 +180,7 @@ dpaa2_eth_set_link_ksettings(struct net_device *net_dev,
 		cfg.options &= ~DPNI_LINK_OPT_HALF_DUPLEX;
 
 	if (dpaa2_eth_cmp_dpni_ver(priv, DPNI_LINK_AUTONEG_VER_MAJOR,
-				   DPNI_LINK_AUTONEG_VER_MINOR)) {
+				   DPNI_LINK_AUTONEG_VER_MINOR) < 0) {
 		err = dpni_set_link_cfg(priv->mc_io, 0, priv->mc_token, &cfg);
 	} else {
 		link_mode_ethtool2dpni(link_settings->link_modes.advertising,
-- 
2.17.1

